tests/conftest.py

```diff
@@ -12,3 +16,25 @@ def change_test_dir(request, monkeypatch):
     :param monkeypatch: -
     """
     monkeypatch.chdir(request.fspath.dirname)
+
+
+@pytest.fixture
+def tmp_dir_factory() -> Generator[Callable[[str], str], None, None]:
```

Comment:
For my information/learning; why do you use a custom `tmp_dir_factory` function? Why not use the built-in [`tmpdir` of pytest](https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmpdir-fixture)? That is what we do in `parcels` (e.g. [here](https://github.com/OceanParcels/parcels/blob/3652e364d2a424ca0ddeea3109ddb5abad91d6d1/tests/test_particlefile.py#L41))

Reply:
I was unaware of its existance and have replaced